chia/simulator/block_tools.py

```diff
@@ -292,6 +296,16 @@ def test_callback(event: PlotRefreshEvents, update_result: PlotRefreshResult) ->
             match_str=str(self.plot_dir.relative_to(DEFAULT_ROOT_PATH.parent)) if not automated_testing else None,
         )
 
+    def load_derivation_cache(self, path: Path) -> None:
+        try:
+            with path.open(encoding="UTF-8") as cache_file:
+                objects = json.load(cache_file)
+                for ob in objects:
+                    a, b, c, d = ob
+                    self.derivation_cache[(bytes.fromhex(a), b, c)] = bytes.fromhex(d)
```

Comment:
wouldn't it be simpler to use pickle here?

Reply:
Yes, definitely! I'll make that change.

Reply:
Hmm, it might be better to have the cached derivations checked in as plaintext, though.